infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerAdapter.java

```diff
@@ -5,47 +5,26 @@
 import org.corfudb.infrastructure.logreplication.infrastructure.CorfuReplicationDiscoveryServiceAdapter;
 import org.corfudb.infrastructure.logreplication.proto.LogReplicationClusterInfo.TopologyConfigurationMsg;
 
-public abstract class CorfuReplicationSiteManagerAdapter {
-    @Getter
-    CorfuReplicationDiscoveryServiceAdapter corfuReplicationDiscoveryService;
-
-    @Getter
-    TopologyConfigurationMsg topologyConfig;
-
-    public void setCorfuReplicationDiscoveryService(CorfuReplicationDiscoveryServiceAdapter corfuReplicationDiscoveryService) {
-        this.corfuReplicationDiscoveryService = corfuReplicationDiscoveryService;
-        start();
-    }
-
-    public synchronized TopologyConfigurationMsg fetchTopology() {
-        topologyConfig = queryTopologyConfig();
-        return topologyConfig;
-    }
-
-    /**
-     * Will be called when the cluster change and a new configuration is sent over
-     *
-     * @param newTopologyConfigMsg
-     */
-    public synchronized void updateTopologyConfig(TopologyConfigurationMsg newTopologyConfigMsg) {
-        if (newTopologyConfigMsg.getTopologyConfigID() > topologyConfig.getTopologyConfigID()) {
-            topologyConfig = newTopologyConfigMsg;
-            corfuReplicationDiscoveryService.updateSiteConfig(topologyConfig);
-        }
-    }
-
-    public void prepareSiteRoleChange() {
-        corfuReplicationDiscoveryService.prepareSiteRoleChange();
-    }
-
-    public int queryReplicationStatus() {
-        return corfuReplicationDiscoveryService.queryReplicationStatus();
-    }
-
-    //TODO: handle the case that queryTopologyConfig return an exception.
-    public abstract TopologyConfigurationMsg queryTopologyConfig();
-
-    public abstract void start();
-
-    public abstract void shutdown();
+public interface CorfuReplicationSiteManagerAdapter {
```

Comment:
Site -> Cluster

Reply:
I am really confused with all the new naming. Corfu is a cluster, but we don't name anything there as cluster. If we follow corfu's naming, we should not introduce cluster in the naming. 
When naming as ClusterManager, it gives the conception of it is manager the whole cluster, but it only gets information for this local node to process.

Reply:
no Corfu is not a Cluster. Corfu is the friendly name of our Datastore. 
I was referring to name it CorfuReplicationClusterManagerAdapter... We agreed that Site is not a 'Corfu' terminology, we're broader in scope than a Site, we talk in terms of Clusters (i.e., Corfu Clusters). 
If you still have questions we can discuss it offline.